docs/block-processing/us.md

```diff
@@ -0,0 +1,359 @@
+    * Otherwise it must be in `X` or greater than maximum from `X` by `1`.
+
+* _BlockVersionModifier consistenty check_: `BlockVersionModifier` from proposal is consistent 
+with `BlockVersionData` for adopted protocol version or with `BlockVersionModifier` from _competing_ proposals.  
+There are three cases to check:
+  * Block version from proposal is equal to adopted version. 
+    In this case each field of `BlockVersionModifier` from proposal must be either `Nothing` or 
+    `Just` with the same value from `BlockVersionData` of last adopted version.
+  * Block version from proposal is equal to one from another proposal. 
+    `BlockVersionModifier`s from them must be the same. 
+     If several proposals with the same block version exist, then all `BlockVersionModifier` must be equal.
+  * Block version from proposal is new one. 
+    In this case we must check `BlockVersionModifier` from the proposal is consistent with adopted `BlockVersionData`.  
+    The following checks take place:
+    * script version from proposal must be equal to or greater by `1` adopted one.
+    * max block size from proposal is at most two times greater than adopted one.
+    * unlock stake epoch must not be changed if boostrap era is over
+
+### Votes verification
+
+For each vote the following checks must be performed:
+
+* _Voter stake check_: stakeholder who issued given vote has enough stake (at least `bvdUpdateProposalThd`).
+  Note: `bvdUpdateProposalThd` is used by mistake, `bvdUpdateVoteThd` must be used. It will be resolved in the future.
+
+* _Existence proposal check_: proposal with such proposal id exists.
+
+* _Active proposal check_: proposal is still active.
+
+* _Revote check_: a stakeholder who issued the vote doesn't revote twice or send the same decision twice.  
+  It's allowed for any stakeholder to vote no more than one time and 
+  also change his decision to opposite no more than one time.
+
+## Confirmation and adoption algorithms
+
+In the previous sections we touched _explicit agreement rule_ and _adoption of block version_,
+we will describe them in details in this section.
+
+### Update proposal confirmation algorithm
+In this section we describe update proposal confirmation algorithm more carefully, 
+in particular, the rules for cases when there are several concurrent update proposals.
+
+On each main block the following algorithm is performed
+1. Take all proposals which were decided at least `k` block ago.  
+   It may be determined by chain difficulty.
+2. Group proposals by application name.
+3. Sort each group using the following rule:
+  * Construct tuple  
+    (decision, whether decision is implicit, positive stake, slot when it has been proposed)  
+    by each proposal. So a tuple has type `(Bool, Bool, Coin, SlotId)`
+  * Compare in decreasing order.  
+    So at the beginning of sorted list,  _approved_ proposals will follow, if there is more than one,
+    approved explicitly and so on.
```

Comment:
I am wondering if it's theoretically possible that all 4 values in tuple will be the same 🤔